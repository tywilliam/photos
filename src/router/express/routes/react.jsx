// @flow

import { Router } from 'express';
import { readFileSync } from 'fs';
import { NOT_FOUND, OK } from 'http-status-codes';
import { join } from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { matchPath, StaticRouter } from 'react-router-dom';
import { ServerStyleSheet } from 'styled-components';

import { App } from '../../../components';
import routes from '../../../router/react';
import { cssAssets, hash, jsAssets } from '../../../utils/webpack-stats';


const router = new Router();

router.get('*', ({ url }, { status }) => {
  const match = routes.reduce((acc, route) => (
    matchPath(url, route, {
      exact: route.extact,
    }) || acc
  ), null);

  if (!match) {
    status(NOT_FOUND).send('Not found');
    return;
  }

  const context = {};

  const stylesheet = new ServerStyleSheet();
  const markup = renderToString(stylesheet.collectStyles(
    <StaticRouter
      context={context}
      location={url}
    >
      <App />
    </StaticRouter>,
  ));
  const css = stylesheet.getStyleTags();

  const manifestPath: string = join(__dirname, jsAssets.find(({ name }) => name === 'manifest').localPath);
  const manifest = readFileSync(manifestPath);

  status(OK).render('react', {
    jsassets: jsAssets.filter(({ name }) => name !== 'manifest'),
    cssassets: cssAssets,
    css,
    hash,
    manifest,
    markup,
    title: '',
  });
});

export { router as default };
