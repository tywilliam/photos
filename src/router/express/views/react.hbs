<div id="root">{{{ markup }}}</div>
