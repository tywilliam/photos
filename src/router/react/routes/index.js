import home from './home';
import photo from './photo';

export home from './home';
export photo from './photo';

export default [
  home,
  photo,
];
