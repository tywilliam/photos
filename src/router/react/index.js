import routes from './routes';

export { routes as default };
