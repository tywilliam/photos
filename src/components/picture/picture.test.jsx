/* global render:false, shallow:false */

import * as React from 'react';

import Picture from './';
import Source from '../source';


describe('<Picture />', () => {
  it('always renders one <img /> tag', () => {
    const wrapper = render(<Picture />);
    expect(wrapper.find('img')).toHaveLength(1);
  });

  xit('only renders <Source /> as children', () => {
    // const wrapper = shallow((
    //   <Picture>
    //     <div className="unique" />
    //   </Picture>
    // ));

    const wrapper = shallow((
      <Picture>
        <Source />
      </Picture>
    ));

    // expect(pictureWithInvalidChild.contains(<div className="unique" />)).toBeFalsy();
    expect(wrapper.find(Source)).toHaveLength(1);
  });
});
