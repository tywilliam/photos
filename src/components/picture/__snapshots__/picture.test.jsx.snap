// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture It renders 1`] = `
<picture
  className="sc-bdVaJa bDWFJH"
>
  <img
    alt={undefined}
    src={undefined}
    title={undefined}
  />
</picture>
`;
